pytorch_lightning/metrics/functional/image_gradients.py

```diff
@@ -0,0 +1,73 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple
+
+import torch
+
+
+def _image_gradients_validate(img: torch.Tensor) -> torch.Tensor:
+
+    if not isinstance(img, torch.Tensor):
+        raise TypeError(f"`img` expects a value of <torch.Tensor> type but got {type(img)}")
+    if img.ndim != 3 and img.ndim != 4:
+        raise RuntimeError(f"`img` expects a 3D or 4D tensor but got {img.ndim}D tensor")
+
+
+def _compute_image_gradients(img: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+
+    batch_size, channels, height, width = img.shape
```

Comment:
I assume you need to `unsqueeze` a dimension if the input is a 3D tensor for this to work? Do you test for 3D tensors? Else it is okay that input needs to be a 4D tensor.

Reply:
I assumed that since this function would generally be used in conjunction with loss-functions or during training, it makes more sense to just accept 4D tensors.

Reply:
That is fine by me, but the `_image_gradients_validate` function does allow for 3D input. IMO it would make sense to only check for 4D tensors

Reply:
Sure I'll push a commit to remove this inconsistency.

Reply:
Pushed a commit fixing all the issues.